developer-resources/tools/IntelliJ.md

```diff
@@ -22,6 +22,8 @@ There is a [tutorial for IntelliJ](../../open-metadata-resources/open-metadata-t
 Lombok Plugin
 
 Egeria makes use of Project Lombok. If using JetBrains IntelliJ IDEA ensure the IDEA has the required plugin configured. See https://projectlombok.org/setup/intellij for more information.
+Also before running a Maven build please choose "Don't detect" from the "Generated sources folders" dropdown in Preferences -> Build, Execution, Deployment -> Build Tools -> Maven -> Importing. This will avoid triggering a duplicate classes build error caused by delomboked sources folder being added as source folder for the Maven module.
```

Comment:
Do you know what the impact to gradle is - our long term goal is still to switch

Reply:
I think this will be something that many people would fall over. If we can have it working with no settings changes - that would be ideal.

Reply:
@planetf1 I tried really fast to import Egeria as a Gradle project in Intellij, but there seem to be some issues while building it after the import. Do you have a working gradle project import?

Otherwise, the gradle build runs successfully as can be seen here too. Anyway, I think the issue is in fact Maven + Intellij + Lombok related.

@davidradl I would like to have it working without a mandatory Intellij setting. Unfortunately, it looks like Intellij has a bug regarding adding the sources in the classpath when running with Lombok: https://youtrack.jetbrains.com/issue/IDEA-184689. The bug is in the "Submitted" state and this is a workaround. I do hope they will fix it soon so we can forget about the settings change.

Reply:
The issue dates from 2018. Could you raise the issue with jetbrains (add to the issue perhaps) to get clarification of expected behaviour?

On gradle - I haven't had time to explore, but I have been able to import egeria as a gradle project directly and without doing further config. However I've kept it in a seperate dir tree as otherwise intellij seems to get confused about whether the project is maven or gradle. But note that in that dir tree do a mvn clean install first (CLI). Why? Because only maven downloads some derby/gaian stuff that the build needs. that is addressed in a PR about to be merged.

Reply:
Worked with @alexandra-bucur on more checks today.
 - If we're going to use lombok and maven I don't think we can escape from this issue - which is really intellij being a little over zealous in looking for generated code . Having researched further I don't think we can do any better, and @alexandra-bucur 's proposal is the best we have.
 - We're not seeing this issue in gradle which is building cleanly at CLI and in INtelliJ. Javadoc & sources are correctly generated including aggregated javadoc . No configuration is needed (this is good news for when we switch to gradle). This is probably due to the better gradle/intellij integration.